modules/products/bazel/src/main/kotlin/com/metalbear/mirrord/products/bazel/BazelExecutionListener.kt

```diff
@@ -0,0 +1,154 @@
+
+        val state = getBlazeConfigurationState(env) ?: run {
+            MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: Bazel not detected")
+            super.processStartScheduled(executorId, env)
+            return
+        }
+        MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: got config state $state")
+
+        MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: wsl check")
+        val wsl = when (val request = createEnvironmentRequest(env.runProfile, env.project)) {
+            is WslTargetEnvironmentRequest -> request.configuration.distribution!!
+            else -> null
+        }
+
+        val originalEnv = state.userEnvVarsState.data.envs
+        MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: found ${originalEnv.size} original env variables")
+        val originalBazelPath = if (SystemInfo.isMac) {
+            state.blazeBinaryState.blazeBinary?.let {
+                MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: found Bazel binary path in the config: $it")
+                it
+            } ?: run {
+                // Bazel binary path may be missing from the config.
+                // This is the logic that Bazel plugin uses to find global Bazel binary.
+                val global = Blaze.getBuildSystemProvider(env.project).buildSystem.getBuildInvoker(env.project, BlazeContext.create()).binaryPath
+                MirrordLogger.logger.debug("[${this.javaClass.name} processStartScheduled: found global Bazel binary path: $global")
+                global
+            }
+        } else {
+            null
+        }
+
+        try {
+            service.execManager.wrapper("bazel", originalEnv).apply {
+                this.wsl = wsl
+                this.executable = originalBazelPath
+            }.start()?.let { executionInfo ->
+                MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: adding ${executionInfo.environment.size} environment variables")
+                var envVars = originalEnv + executionInfo.environment
+                executionInfo.envToUnset?.let { envToUnset ->
+                    envVars = envVars.filter {
+                        !envToUnset.contains(it.key)
+                    }
+                }
+                state.userEnvVarsState.setEnvVars(envVars)
+
+                if (SystemInfo.isMac) {
+                    MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: isMac, patching SIP.")
+                    executionInfo.patchedPath?.let {
+                        MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: patchedPath is not null: $it, meaning original was SIP")
+                        state.blazeBinaryState.blazeBinary = it
+                    }
+                }
+
+                savedEnvs[executorId] = SavedConfigData(originalEnv, originalBazelPath)
```

Comment:
If `blazeBinary` was originally null, `originalBazelPath` here we'll be the global path here, right?
Shouldn't it be null instead in that case, so that we restore the original state after we're done?